### PR TITLE
Add support for visualizing v8 profile 

### PR DESCRIPTION
**Pull request by slonka**
...gathered by CpuProfiler::StartProfling.

As suggested in the comment: https://github.com/davidmarkclements/0x/issues/196#issuecomment-436196449 here is an example https://github.com/slonka/0x-visualize-v8-profile-example of how to generate v8 profile at runtime and a flamegraph from it.

Example flamegraph looks like this:
screen shot 2018-11-06 at 16 36 15

There is a couple of things that this PR is missing:
- [x] tests
- [x] merged / unmerged
- [x] optimized / unoptimized could probably be taken from "bailout reason"
- [ ] I do not know how 0x detects tiers

But the basic functionality is there. Would love help improving this.

### CODE REVIEW

File: lib/convert-v8-profile-to-tree.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+function convert (profile) {
```
Comment (mcollina) [inline lib/convert-v8-profile-to-tree.js]: Can you add `'use strict'`  at the top?

**Comment by slonka** [comment]
@mcollina you just go into the example repo I linked to  https://github.com/slonka/0x-visualize-v8-profile-example, clone it, run `npm i` and `./test.sh` and you can see the `flamegraph.html` in the folder

**Comment by mcollina** [comment]
Can you add that code in the https://github.com/davidmarkclements/0x/tree/master/examples and add a section about it in the README?

**Comment by slonka** [comment]
Yeah sure, I thought that you had problems running the example.

File: readme.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ on any platform which Node runs on (macOs, Linux, Windows, Android...).
 * Node v8.5.0 and above
 * Default usage supports any Operating System that Node runs on!
 * Chrome
-  * Other browsers may open flamegraphs in a degraded, but functional form 
+  * Other browsers may open flamegraphs in a degraded, but functional form
```
Comment (slonka) [inline readme.md]: I did not change it, some formatter did.

**Comment by mcollina** [comment]
I think you can’t provide that info through that output. We can detect that by parsing the isolate.log file.

The buttons can either do nothing, or we can grey them out in this case.

**Comment by slonka** [comment]
Yes, sure. How can I disable the "merge" button?

**Comment by davidmarkclements** [comment]
if the v8-profiler doesn't differentiate between optimized/unoptimized we'll also need to disable the Optimized Unoptimized buttons as well

Additionally the filter buttons – possibly many of those don't apply? They should also be disabled


The buttons are HTML buttons so can have a disabled attribute set, this is passed down to a basic function component here: 

https://github.com/davidmarkclements/0x/blob/master/visualizer/cmp/controls.js#L3-L15

and they're called here: https://github.com/davidmarkclements/0x/blob/master/visualizer/cmp/controls.js#L19-L33


you might need to adjust/add to the state (https://github.com/davidmarkclements/0x/blob/master/visualizer/state.js) to determine whether a button should be disabled, 

the state is initialized in the visualizer here: https://github.com/davidmarkclements/0x/blob/master/visualizer/index.js#L52

and options can be passed to the visualizer from here: https://github.com/davidmarkclements/0x/blob/master/lib/render.js#L23

**Comment by slonka** [comment]
@davidmarkclements - I disabled the buttons when flamegraph is generated from v8 profiler output.

It looks like this:
screen shot 2018-12-04 at 22 15 00

**Comment by slonka** [comment]
BTW, this repo needs more tests definitely because the previous version broke the tool and I only noticed when I wanted to use it with different switches. I'll open a new PR for that.

**Comment by mcollina** [comment]
@slonka some tests have been added in the last week. You might want to rebase.

**Comment by slonka** [comment]
I did the rebase, any chance to get this merged? @mcollina @davidmarkclements

**Comment by davidmarkclements** [comment]
amazing PR @slonka sorry it took so long, I'm just going to rename it to --visualize-cpu-profile – especially in light of https://github.com/nodejs/node/pull/27147

**Comment by davidmarkclements** [comment]
renaming commit here:
https://github.com/davidmarkclements/0x/commit/d5c8d3b36bd59f7188be9c3655885a0b265489f7

**Comment by slonka** [comment]
Thanks for merging this 🎊

**Comment by slonka** [comment]
Are you sure the format of --cpu-prof matches the one provided here?

**Comment by davidmarkclements** [comment]
No, however I expect it to be very similar to the extent that a tweak or two will make it work for both scenarios

**Pull request merged**